ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1432,14 +1432,19 @@ private class JdbcLedgerDao(
     )
 
   override def getParties: Future[List[PartyDetails]] =
-    dbDispatcher.executeSql("load_parties") { implicit conn =>
-      SQL_SELECT_PARTIES
-        .as(PartyDataParser.*)
-        // TODO: isLocal should be based on equality of participantId reported in an
-        // update and the id given to participant in a command-line argument
-        // (See issue #2026)
-        .map(d => PartyDetails(Party.assertFromString(d.party), d.displayName, isLocal = true))
-    }
+    dbDispatcher
+      .executeSql("load_parties") { implicit conn =>
+        SQL_SELECT_PARTIES
+          .as(PartyDataParser.*)
+      }
+      .map(
+        _.map(
+          d =>
+            // TODO: isLocal should be based on equality of participantId reported in an
+            // update and the id given to participant in a command-line argument
+            // (See issue #2026)
+            PartyDetails(Party.assertFromString(d.party), d.displayName, isLocal = true)))(
+        executionContext)
```

Comment:
Looks like this `executionContext` comes from `actorSystem.dispatcher`, which comes from the `StandaloneIndexer`'s actor system. Just wanted to make sure we're OK with this.

Reply:
I agree we should change this, but in the context of this object (as it is right now) the `executionContext` is where `Future`s should be ran. I'll probably have a go at this as part of my participant redesign story.

Thanks for pointing this out though, as mentioned I agree that we could do a better job at organizing our thread pools.